crates/planus/src/impls/primitives.rs

```diff
@@ -98,13 +102,13 @@ macro_rules! gen_primitive_types_with_vectors {
     };
 }
 
-gen_primitive_types!(i8, 1);
-gen_primitive_types!(u8, 1);
-gen_primitive_types_with_vectors!(i16, 2);
-gen_primitive_types_with_vectors!(u16, 2);
-gen_primitive_types_with_vectors!(i32, 4);
-gen_primitive_types_with_vectors!(u32, 4);
-gen_primitive_types_with_vectors!(i64, 8);
-gen_primitive_types_with_vectors!(u64, 8);
-gen_primitive_types_with_vectors!(f32, 4);
-gen_primitive_types_with_vectors!(f64, 8);
+unsafe_gen_primitive_types!(i8, core::mem::size_of::<i8>());
```

Comment:
This is pedantic given https://doc.rust-lang.org/reference/type-layout.html#primitive-data-layout but in general I think it is a more solid pattern should someone try to copy that macro call

Reply:
I think this code might be from before `size_of` was a const function, and that's why it looks like that? In any LGTM.

Reply:
Indeed. The MSRV for it is 1.24 so I am going to stick with it.

I will not change the signature because that is an unnecessary breaking change. `$size` could always be that but let's not break downstream users even more